test/e2e/discovery/statefulsets.go

```diff
@@ -0,0 +1,269 @@
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	lhconstants "github.com/submariner-io/lighthouse/pkg/constants"
+	lhframework "github.com/submariner-io/lighthouse/test/e2e/framework"
+	"github.com/submariner-io/shipyard/test/e2e/framework"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/api/discovery/v1beta1"
+)
+
+var _ = Describe("[discovery] Test Stateful Sets Discovery Across Clusters", func() {
+	f := lhframework.NewFramework("discovery")
+
+	When("a pod tries to resolve a podname from stateful set in a remote cluster", func() {
+		It("should resolve the pod IP from the remote cluster", func() {
+			if !framework.TestContext.GlobalnetEnabled {
+				RunSSDiscoveryTest(f)
+			} else {
+				framework.Skipf("Globalnet is enabled, skipping the test...")
+			}
+		})
+	})
+
+	When("a pod tries to resolve a podname from stateful set in a local cluster", func() {
+		It("should resolve the pod IP from the local cluster", func() {
+			if !framework.TestContext.GlobalnetEnabled {
+				RunSSDiscoveryLocalTest(f)
+			} else {
+				framework.Skipf("Globalnet is enabled, skipping the test...")
+			}
+		})
+	})
+
+	When("the number of active pods backing a stateful set changes", func() {
+		It("should only resolve the IPs from the active pods", func() {
+			if !framework.TestContext.GlobalnetEnabled {
+				RunSSPodsAvailabilityTest(f)
+			} else {
+				framework.Skipf("Globalnet is enabled, skipping the test...")
+			}
+		})
+	})
+})
+
+func RunSSDiscoveryTest(f *lhframework.Framework) {
+	clusterAName := framework.TestContext.ClusterIDs[framework.ClusterA]
+	clusterBName := framework.TestContext.ClusterIDs[framework.ClusterB]
+
+	By(fmt.Sprintf("Creating an Nginx Stateful Set on on %q", clusterBName))
+
+	nginxSSClusterB := f.NewNginxStatefulSet(framework.ClusterB)
+	appName := nginxSSClusterB.Spec.Selector.MatchLabels["app"]
+
+	By(fmt.Sprintf("Creating a Nginx Headless Service on %q", clusterBName))
+
+	nginxServiceClusterB := f.NewNginxHeadlessServiceWithParams(nginxSSClusterB.Spec.ServiceName, appName, framework.ClusterB)
+
+	f.NewServiceExport(framework.ClusterB, nginxServiceClusterB.Name, nginxServiceClusterB.Namespace)
+	f.AwaitServiceExportedStatusCondition(framework.ClusterB, nginxServiceClusterB.Name, nginxServiceClusterB.Namespace)
+
+	By(fmt.Sprintf("Creating a Netshoot Deployment on %q", clusterAName))
+
+	netshootPodList := f.NewNetShootDeployment(framework.ClusterA)
+
+	endpointSlices := f.AwaitEndpointSlices(framework.ClusterB, nginxServiceClusterB.Name, nginxServiceClusterB.Namespace, 1, 1)
+
+	verifyCount := 0
+
+	for _, endpointSlice := range endpointSlices.Items {
+		sourceCluster := endpointSlice.Labels[lhconstants.LabelSourceCluster]
+
+		for _, endpoint := range endpointSlice.Endpoints {
+			verifyEndpointsWithDig(f.Framework, framework.ClusterA, netshootPodList, endpoint, sourceCluster,
+				nginxServiceClusterB.Name, checkedDomains, true)
+			verifyCount++
+		}
+	}
+
+	Expect(verifyCount).To(Equal(1))
```

Comment:
When this fails it won't be very descriptive, I suggest we add some meaningful explanation in the expect:
```
	Expect(verifyCount).To(Equal(1), "Explaining message...")
```